### DISCUSSION TITLE
Trace duration and "off-span" durations as metrics

### DISCUSSION THREAD

**Discussion by AlexSilver9** [open-telemetry/community#1825] [Community Discussions]
Hi,

In tracing tools like Jaeger I can see the overall duration of a trace from start to end.
I need the same as metric visible in prometheus to generate alerts based on it.
Is there a way to get the duration of traces as metrics?

Using the spanmetricsconnector I can see the durations of spans as metrics, which in my context are durations my services need to process events.
What I also need is the duration between the spans, e.g. for how long are events in kafka or other infrastructure components.
Is there a way to get the "off-span"-durations as metrics?

Thank you!

**Comment by trask** [comment]
hi @AlexSilver9!

Do these experimental attributes in Java instrumentation address what you're looking for? https://github.com/search?q=repo%3Aopen-telemetry%2Fopentelemetry-java-instrumentation%20queue_time_ms&type=code

I'd recommend opening an issue in https://github.com/open-telemetry/semantic-conventions, and we can get the messaging folks to chime in further.

**Reply by AlexSilver9** [reply]
Wow! What a delay, just got notified about your answer some days ago. Thank you for your comment. Unfortunately I am no more active on that task since a long time, and can't share any insights.
